system/jlib/jmisc.hpp

```diff
@@ -272,10 +277,12 @@ interface IAbortRequestCallback
 class LocalAbortHandler
 {
 public:
-    LocalAbortHandler(AbortHandler _handler=NULL)   { handler = _handler; addAbortHandler(handler); }
-    ~LocalAbortHandler()                            { removeAbortHandler(handler); }
+    LocalAbortHandler(AbortHandler _handler=NULL)   { handler = _handler; shandler = NULL; addAbortHandler(handler); }
+    LocalAbortHandler(SimpleAbortHandler _handler=NULL) { shandler = _handler; handler = NULL; addAbortHandler(shandler); }
+    ~LocalAbortHandler()                            { if (handler) { removeAbortHandler(handler); } else removeAbortHandler(shandler); }
```

Comment:
@ghalliday went ahead and added in the removeAbortHandler(shandler) code.  Thanks for catching that!